src/dspaces-conf.c

```diff
@@ -313,10 +313,63 @@ static void parse_modules_table(toml_table_t *modules, struct ds_conf *conf)
 #endif
 }
 
+static void parse_swap_table(toml_table_t *swap, struct ds_conf *conf)
+{
+    toml_datum_t dat;
+
+    dat = toml_string_in(swap, "directory");
```

Comment:
The `get_toml_str()` doesn't handle the no-key situation. For the swap toml table, we have the default settings and don't force user to specify it in the configuration file, so I'll stick to this native function call at this time.

Reply:
`get_toml_str()` doesn't alter the value if the key doesn't exist. So

```
s = strdup("foo");
get_toml_str(table, "bar", &s)
```

Leaves s alone if there is no `bar` in `table`, and sets it otherwise. It does memory leak if it sets it though.

It's fine to leave it as is, but it would be nice to refactor eventually.

Reply:
Right, the memory leak is my concern given that I'm trying to `set_default_conf()` first and then read the strings. I'll leave it as it is now.